src/goto-checker/multi_path_symex_checker.cpp

```diff
@@ -155,3 +166,13 @@ multi_path_symex_checkert::localize_fault(const irep_idt &property_id) const
 
   return fault_localizer(property_id);
 }
+
+void multi_path_symex_checkert::report()
+{
+  if(options.is_set("write-solver-stats-to"))
+  {
+    with_solver_hardness(property_decider, [](solver_hardnesst &hardness) {
```

Comment:
pass decision procedure instead of property decider:
`with_solver_hardness(property_decider.get_decision_procedure()).produce_report();`